engine/src/main/java/org/terasology/world/block/loader/BlockFamilyDefinition.java

```diff
@@ -48,17 +51,25 @@ protected void doReload(BlockFamilyDefinitionData blockFamilyDefinitionData) {
     }
 
     public boolean isFreeform() {
```

Comment:
isFreeform is a bit strange in this context. never really quite understood the reasoning and how this fit into blockfamilies. at this point it just checks if the block has shapes defined for it. would it be worth deprecating and making getShapes return Optional<ArrayList<BlockShape>>.

Reply:
Returning an Optional in this case wouldn't be better than just optionally returning null. I could just inline this method (it isn't used in many places), but I just kept it like this because block family definitions with the shapes attribute are treated a bit differently from ones without that attribute (the family can vary according to the shape, so for blocks with multiple shapes, the family is given less control over the construction of the block), so it seemed appropriate.

Reply:
That makes sense.

Reply:
What's the difference between an empty list and `null`/`Optional.empty()`?